.github/workflows/build.yaml

```diff
@@ -9,6 +9,7 @@ on:
     branches:
       - main
       - 'integration/**'
+  workflow_dispatch:
```

Comment:
Is this enough to let you run this workflow for any commit?  A quick search of the docs makes me think it can run for different branches: https://docs.github.com/en/actions/how-tos/manage-workflow-runs/manually-run-a-workflow